akka-http2-support/src/main/scala/akka/http/scaladsl/Http2.scala

```diff
@@ -180,6 +180,29 @@ final class Http2Ext(private val config: Config)(implicit val system: ActorSyste
     ProtocolSwitch(_ => getChosenProtocol(), http1, http2) join
       tls
   }
+
+  def outgoingConnection(
```

Comment:
Already has same signature as the existing API in `Http()`. Due to multiplexing the stream has different semantics than for HTTP/1.1, so it's more similar in usage as the `newHostConnectionPool` API in `Http()`.

For the public APIs we have to consider how to integrate it. (Will create a ticket about it)